### PR TITLE
feat(KIC): add reference page for resources requiring ingress class

### PR DESCRIPTION
**Pull request by randmonkey**
### Description

 

Add reference page to introduce resources that requires setting ingress classes to get reconciled by KIC.
Fixes https://github.com/Kong/kubernetes-ingress-controller/issues/4964. 

### Testing instructions

Preview link: 
https://deploy-preview-6480--kongdocs.netlify.app/kubernetes-ingress-controller/latest/reference/resource-requiring-ingress-class/
### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

Note for reviewer: Should we add links to each resource in the doc?

### CODE REVIEW

File: app/_src/kubernetes-ingress-controller/reference/ingress-class-requirement.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  What resources requires setting ingress class by its field or annotation to get reconciled by {{site.kic_product_name}} and how to set them
+---
+
+## Ingress Class
+
+[Ingress class][ingress_class] is defined to distinguish ingresses to be processed by different controllers. {{site.kic_product_name}} has the flag `--ingress-class` (default:`kong`) to set the ingress class that the {{site.kic_product_name}} instance processes. In gateway API, [gateway class][gateway_class] has the same usage.
+
+## Resource Requiring Setting Ingress Class or Gateway Class
+
+Some resources requires setting ingress class to get processed by {{site.kic_product_name}}. The ingress class need to be set to the same as specified in the flag `--ingress-class` to get processed. The table goes below.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/reference/ingress-class-requirement.md:14]:
```suggestion
Some resources requires setting ingress class to get processed by {{site.kic_product_name}}. The ingress class need to be set to the same as specified in the flag `--ingress-class` to get processed. Here's the table with the details.
```

File: app/_src/kubernetes-ingress-controller/reference/ingress-class-requirement.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+| `Ingress`           | `networking.k8s.io/v1`             | Set in `spec.ingressClassName` or               |
+|                     |                                    | set in annotation `kubernetes.io/ingress.class` |
+| `Gateway`           | `gateway.networking.k8s.io/v1`     | Set in `spec.gatewayClassName`                  |
+| `KongClusterPlugin` | `configuration.konghq.com/v1`      | Set in annotation `kubernetes.io/ingress.class` |
+| `KongConsumer`      | `configuration.konghq.com/v1`      | Set in annotation `kubernetes.io/ingress.class` |
+| `KongConsumerGroup` | `configuration.konghq.com/v1beta1` | Set in annotation `kubernetes.io/ingress.class` |
+| `TCPIngress`        | `configuration.konghq.com/v1beta1` | Set in annotation `kubernetes.io/ingress.class` |
+| `UDPIngress`        | `configuration.konghq.com/v1beta1` | Set in annotation `kubernetes.io/ingress.class` |
+
+Note: If the `IngressClass` used by {{site.kic_product_name}} (specified in flag `--ingress-class`) has `ingressclass.kubernetes.io/is-default-class` set to `true`, the resources except for `Gateway` does not have ingress classes set will also get reconciled by {{site.kic_product_name}}.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/reference/ingress-class-requirement.md:27]:
```suggestion
Note: If the `IngressClass` used by {{site.kic_product_name}} (specified in flag `--ingress-class`) has `ingressclass.kubernetes.io/is-default-class` set to `true`, the resources except for `Gateway` does not have ingress classes set are also reconciled by {{site.kic_product_name}}.
```

**Review by Rajakavitha1 — APPROVED** [review]
LGTM

File: app/_src/kubernetes-ingress-controller/reference/ingress-class-requirement.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+| `Ingress`           | `networking.k8s.io/v1`             | Set in `spec.ingressClassName` or               |
+|                     |                                    | set in annotation `kubernetes.io/ingress.class` |
+| `Gateway`           | `gateway.networking.k8s.io/v1`     | Set in `spec.gatewayClassName`                  |
+| `KongClusterPlugin` | `configuration.konghq.com/v1`      | Set in annotation `kubernetes.io/ingress.class` |
+| `KongConsumer`      | `configuration.konghq.com/v1`      | Set in annotation `kubernetes.io/ingress.class` |
+| `KongConsumerGroup` | `configuration.konghq.com/v1beta1` | Set in annotation `kubernetes.io/ingress.class` |
+| `TCPIngress`        | `configuration.konghq.com/v1beta1` | Set in annotation `kubernetes.io/ingress.class` |
+| `UDPIngress`        | `configuration.konghq.com/v1beta1` | Set in annotation `kubernetes.io/ingress.class` |
+
+Note: If the `IngressClass` used by {{site.kic_product_name}} (specified in flag `--ingress-class`) has `ingressclass.kubernetes.io/is-default-class` set to `true`, the resources except for `Gateway` does not have ingress classes set are also reconciled by {{site.kic_product_name}}.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/reference/ingress-class-requirement.md:27]:
```suggestion
**Note**: If the `IngressClass` used by {{site.kic_product_name}} (specified in flag `--ingress-class`) has `ingressclass.kubernetes.io/is-default-class` set to `true`, all the resources that does not have ingress classes set are also reconciled by {{site.kic_product_name}}, except for `Gateway`.
```

File: app/_src/kubernetes-ingress-controller/reference/ingress-class-requirement.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  What resources requires setting ingress class by its field or annotation to get reconciled by {{site.kic_product_name}} and how to set them
+---
+
+## Ingress Class
+
+[Ingress class][ingress_class] is defined to distinguish ingresses to be processed by different controllers. {{site.kic_product_name}} has the flag `--ingress-class` (default:`kong`) to set the ingress class that the {{site.kic_product_name}} instance processes. In gateway API, [gateway class][gateway_class] has the same usage.
+
+## Resource Requiring Setting Ingress Class or Gateway Class
+
+Some resources requires setting ingress class to get processed by {{site.kic_product_name}}. The ingress class need to be set to the same as specified in the flag `--ingress-class` to get processed. Here's the table with the details.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/reference/ingress-class-requirement.md:14]:
```suggestion
Some resources requires that you set ingress class to be processed by {{site.kic_product_name}}. Set the ingress class to the same as specified in the flag `--ingress-class` to get processed. Here's the table with the details.
```

File: app/_src/kubernetes-ingress-controller/reference/ingress-class-requirement.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+---
+title: Resources Requiring Setting Ingress Class and How to Set them
+type: reference
+purpose: |
+  What resources requires setting ingress class by its field or annotation to get reconciled by {{site.kic_product_name}} and how to set them
+---
+
+## Ingress Class
+
+[Ingress class][ingress_class] is defined to distinguish ingresses to be processed by different controllers. {{site.kic_product_name}} has the flag `--ingress-class` (default:`kong`) to set the ingress class that the {{site.kic_product_name}} instance processes. In gateway API, [gateway class][gateway_class] has the same usage.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/reference/ingress-class-requirement.md:10]:
```suggestion
[Ingress class][ingress_class] is defined to distinguish ingresses to be processed by different controllers. {{site.kic_product_name}} has the flag `--ingress-class` and by default it is set to `kong` for the {{site.kic_product_name}} instance processes. In gateway API, [gateway class][gateway_class] has a similar usage.
```

**Review by Rajakavitha1 — APPROVED** [review]
LGTM

**Comment by randmonkey** [comment]
@Rajakavitha1 addressed the comments, please review again.

**Review by mheap — APPROVED** [review]

**Pull request merged**